src/Http/Http.Features/src/IResponseCookies.cs

```diff
@@ -23,6 +25,13 @@ public interface IResponseCookies
         /// <param name="options"><see cref="CookieOptions"/> included in the new cookie setting.</param>
         void Append(string key, string value, CookieOptions options);
 
+        /// <summary>
+        /// Add elements of specified dictionary as cookies.
+        /// </summary>
+        /// <param name="keyValuePairs">Key value pair collections whose elements will be added as cookies.</param>
+        /// <param name="options"><see cref="CookieOptions"/> included in new cookie settings.</param>
+        void Append(IDictionary<string, string> keyValuePairs, CookieOptions options);
```

Comment:
This is a breaking change.

Reply:
Added explanation for this on the PR and the ticket #31508, will be cool to have this but let me know what you think.

Reply:
New methods cannot be added directly to interfaces. The best mitigation in this case might be a DIM.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods

Reply:
Another option is to make it an extension method that knows about `ResponseCookiesWrapper`:

```C#
public static void Append(this IResponseCookies cookies, IReadOnlyDictionary<string, string> values, CookieOptions options)
{
   if (cookies is null) { throw ArgNullException(); }
   if (cookies is ResponseCookieWrapper wrapper)
   {
      wrapper.AppendCookies(values, options);
     return;
   }

   foreach (var item in values) 
   {  
     cookies.Append(item.Key, item, value, options);
   }
}
```

It avoids changes to the interface.

Reply:
We need to move these interfaces to Http

Reply:
@pranavkm how would ResponseCookieWrapper know how to append cookies as a collection to the ResponseCookies type though?

Reply:
Oh fun, didn't realize there's two different implementations. I guess default interface implementation solves that much better, doesn't it?

Reply:
> We need to move these interfaces to Http

@davidfowl we can't move types without a type forward, and we can't forward in that direction without creating a circular dependency.

Why do they need to move? Because Http.Features is a netstandard 2.0 package due to the SignalR client dependency? The client wouldn't need these features so ifdefing them should be fine.

Reply:
> Why do they need to move? Because Http.Features is a netstandard 2.0 package due to the SignalR client dependency? The client wouldn't need these features so ifdefing them should be fine.

I want to split the IFeatureCollection from the features so we can actually change them and so we don't have http features depending on assemblies like this.